includes/wc-template-functions.php

```diff
@@ -895,12 +893,10 @@ function woocommerce_result_count( $query = null ) {
 	/**
 	 * Output the product sorting options.
 	 *
-	 * @param WP_Query $query Pass a query object or use global.
+	 * @param array $args Pass an associative array of parameters. Uses this if passed, otherwise uses global $wp_query.
 	 */
-	function woocommerce_catalog_ordering( $query = null ) {
-		if ( ! is_a( $query, 'WP_Query' ) ) {
-			$query = $GLOBALS['wp_query'];
-		}
+	function woocommerce_catalog_ordering( $args = array() ) {
+		$query = $GLOBALS['wp_query'];
```

Comment:
I'd switch the logic up here somewhat and in the other function so it's more readable.

- If $args is empty, inherit props from global query object.
- Include array of default args and use wp_parse_args
- Within the function, you can then use the variable names after parsing instead of checking isset multiple times. This also let's us whitelist what is used.

Reply:
Also let's you change the variable naming if needed.